### PR TITLE
[WIP] Set tolerance on isometry conversion

### PR DESCRIPTION
**Pull request by JamesSeaward**
### Summary
Added a tolerance to the `Isometry` class so that users have the option to define for themselves instead of stuck with hard-coded value of `1e-10`.

Fixes [https://github.com/Qiskit/qiskit-terra/issues/3789](https://github.com/Qiskit/qiskit-terra/issues/3789)

### Details and comments
The `Isometry` class has a new `eps` attribute that can be passed in during initialisation. 

I could do with some guidance on:
- if the structure of this change "fits" the repos code-style.
- whether or not this needs additional tests (the current `test_isometry.py` file doesn't use the `Isometry` class explicitly anywhere?)
- Where should this change be documented?

### CODE REVIEW

File: qiskit/extensions/quantum_initializer/isometry.py [hunk 1]
```diff
@@ -74,6 +74,7 @@ def __init__(self, isometry, num_ancillas_zero, num_ancillas_dirty):
 
         self.num_ancillas_zero = num_ancillas_zero
         self.num_ancillas_dirty = num_ancillas_dirty
+        self.eps = eps
```
Comment (ajavadia) [inline qiskit/extensions/quantum_initializer/isometry.py]: should probably be `self._eps` (private) so you can't change it after making an Isometry instruction.

File: qiskit/extensions/quantum_initializer/isometry.py [hunk 2]
```diff
@@ -64,7 +64,7 @@ class Isometry(Instruction):
     # finally, we convert the labels back to the qubit numbering used in Qiskit
     # (using: _get_qubits_by_label)
 
-    def __init__(self, isometry, num_ancillas_zero, num_ancillas_dirty):
+    def __init__(self, isometry, num_ancillas_zero, num_ancillas_dirty, eps=1e-10):
```
Comment (ajavadia) [inline qiskit/extensions/quantum_initializer/isometry.py]:
```suggestion
    def __init__(self, isometry, num_ancillas_zero, num_ancillas_dirty, eps=_EPS):
```
for consistency with a bunch of other files which hardcode the precision at the top, yet allow user setting. e.g. see: `qiskit/quantum_info/synthesis/one_qubit_decompose.py`

**Comment by ajavadia** [comment]
@JamesSeaward hi, any update on this?

**Comment by 1ucian0** [comment]
Since this PR is fixing a "good first issue" and seems stalled, Im closing it to allow other newcomers to have a shot. If @JamesSeaward is back, it can be reopen.

**Pull request closed without merging**